test/Microsoft.Framework.DependencyInjection.Tests/ActivatorUtilitiesTests.cs

```diff
@@ -220,5 +232,16 @@ public void GetServiceOrCreateInstanceUnregisteredService()
             Assert.Equal(2, service.InstanceId);
             Assert.Equal(2, CreationCountFakeService.InstanceCount);
         }
+
+        [Fact]
+        public void CreateFactoryUnRegisteredServiceAsConstructorParameter()
```

Comment:
Can we have this same test for `CreateInstance`? I think making this a theory would be better.